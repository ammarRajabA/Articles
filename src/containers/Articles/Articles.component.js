import React, {Component} from 'react';
import { connect } from 'react-redux';

import Spinner from '../../components/Spinner/Spinner.component.js';
import Grid from '../../components/Grid/Grid.component.js';

import {getArticles} from './Articles.actions.js' ;

import history from '../../routes/history'

import './Articles.style.css';

export class Articles extends Component{
	state={}
	componentDidMount(){
		this.props.getArticles()
	}
	renderArticles=()=>{
		if (this.props.articles.loading) return <Spinner/>
		else if (this.props.articles.articles.length===0) return (
			<div className="empty-list">
				<img className="image-placeholder" src="/logo.png"/>
				<span className="empty-label">No articles</span>
			</div>
		)
		else return <Grid items={this.props.articles.articles} cols={3} handleSelect={(item)=>history.push(`/articles/${item.id}`)}/>
	}
	render(){
		return(
			<div className="articles-container">
				{this.renderArticles()}
			</div>
			)
	}
}

const mapStateToProps=(state)=>{
	return {articles:state.articles}
}

export default connect(mapStateToProps,{getArticles})(Articles);