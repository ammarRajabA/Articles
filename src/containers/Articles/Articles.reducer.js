const INIT={
	articles:[],
	loading:false,
}

export default (state=INIT,action)=>{
	switch(action.type){
		case 'START_GET_ARTICLES':
			return {...state,articles:[],loading:true}
		case 'FINISH_GET_ARTICLES':
			return {...state,articles:action.payload,loading:false}
		default:
			return state;
	}
}