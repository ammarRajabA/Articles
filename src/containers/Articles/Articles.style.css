.articles-container{
	height: 100%;
	width: 100%;
}
.articles-container .empty-list{
	display: flex;
	flex-direction: column;
	align-items: center;
	justify-content: center;
	margin-right: auto;
	margin-left: auto;
	width: 50%;
	height: 50%;
}
.articles-container .image-placeholder{
	width: 50%;
}
.articles-container .empty-label{
	font-size: 24px;
	font-style: italic;
	font-family: calibri;
}