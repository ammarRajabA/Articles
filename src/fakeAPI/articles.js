import articles from './fake_articles'
export default {
	get:()=>{
		return {data:articles}
	},
	patch:(id,article)=>{
		var target=articles.findIndex((article,)=>article.id===id);
		articles[target]={...article}
	}
}