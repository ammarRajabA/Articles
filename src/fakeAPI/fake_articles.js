export default[
	{
		title:"first",
		content:JSON.stringify({"blocks":[{"key":"5viej","text":"This is my first article, it so cool I can color words ","type":"unstyled","depth":0,"inlineStyleRanges":[{"offset":49,"length":5,"style":"color-rgb(243,121,52)"}],"entityRanges":[],"data":{}},{"key":"dh0hv","text":"I can also make bold italic or any other style.","type":"unstyled","depth":0,"inlineStyleRanges":[{"offset":16,"length":5,"style":"BOLD"},{"offset":21,"length":7,"style":"ITALIC"}],"entityRanges":[],"data":{}},{"key":"blt1q","text":"I also can write math 25 =32","type":"unstyled","depth":0,"inlineStyleRanges":[{"offset":23,"length":2,"style":"SUPERSCRIPT"}],"entityRanges":[],"data":{}},{"key":"ddclt","text":"ðŸ“£ Attention here : My Github account Ammar Rajab ","type":"unstyled","depth":0,"inlineStyleRanges":[{"offset":2,"length":48,"style":"color-rgb(209,72,65)"}],"entityRanges":[{"offset":38,"length":11,"key":0}],"data":{}}],"entityMap":{"0":{"type":"LINK","mutability":"MUTABLE","data":{"url":"https://github.com/ammarrajaba","targetOption":"_blank"}}}}),
		image:"https://picsum.photos/id/258/200/300",
		author:"Ammar Rajab",
		id:"13245"
	},
	{
		title:"Second",
		content:JSON.stringify({"blocks":[{"key":"35tp4","text":"Another article by me,","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}},{"key":"a1v72","text":"It shows how we align text","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}},{"key":"cifh6","text":"Left","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}},{"key":"b5gsb","text":"Middle","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{"text-align":"center"}},{"key":"eopjr","text":"and right","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{"text-align":"right"}},{"key":"1h6bs","text":"","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}},{"key":"d3mrc","text":"and change font","type":"unstyled","depth":0,"inlineStyleRanges":[{"offset":0,"length":15,"style":"fontfamily-Impact"}],"entityRanges":[],"data":{}}],"entityMap":{}}),
		image:"https://picsum.photos/id/232/200/300",
		author:"Ammar Rajab",
		id:"15555"
	},
	{
		title:"Third",
		content:JSON.stringify({"blocks":[{"key":"5viej","text":"This is my first article, it so cool I can color words ","type":"unstyled","depth":0,"inlineStyleRanges":[{"offset":49,"length":5,"style":"color-rgb(243,121,52)"}],"entityRanges":[],"data":{}},{"key":"dh0hv","text":"I can also make bold italic or any other style.","type":"unstyled","depth":0,"inlineStyleRanges":[{"offset":16,"length":5,"style":"BOLD"},{"offset":21,"length":7,"style":"ITALIC"}],"entityRanges":[],"data":{}},{"key":"blt1q","text":"I also can write math 25 =32","type":"unstyled","depth":0,"inlineStyleRanges":[{"offset":23,"length":2,"style":"SUPERSCRIPT"}],"entityRanges":[],"data":{}},{"key":"ddclt","text":"ðŸ“£ Attention here : My Github account Ammar Rajab ","type":"unstyled","depth":0,"inlineStyleRanges":[{"offset":2,"length":48,"style":"color-rgb(209,72,65)"}],"entityRanges":[{"offset":38,"length":11,"key":0}],"data":{}}],"entityMap":{"0":{"type":"LINK","mutability":"MUTABLE","data":{"url":"https://github.com/ammarrajaba","targetOption":"_blank"}}}}),
		image:"https://picsum.photos/id/258/200/300",
		author:"Ammar Rajab",
		id:"13445"
	},
	{
		title:"Forth",
		content:JSON.stringify({"blocks":[{"key":"35tp4","text":"Another article by me,","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}},{"key":"a1v72","text":"It shows how we align text","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}},{"key":"cifh6","text":"Left","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}},{"key":"b5gsb","text":"Middle","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{"text-align":"center"}},{"key":"eopjr","text":"and right","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{"text-align":"right"}},{"key":"1h6bs","text":"","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}},{"key":"d3mrc","text":"and change font","type":"unstyled","depth":0,"inlineStyleRanges":[{"offset":0,"length":15,"style":"fontfamily-Impact"}],"entityRanges":[],"data":{}}],"entityMap":{}}),
		image:"https://picsum.photos/id/232/200/300",
		author:"Ammar Rajab",
		id:"15005"
	}
]
