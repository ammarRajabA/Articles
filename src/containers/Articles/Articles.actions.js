import articles from '../../fakeAPI/articles'

export const getArticles=()=>dispatch=>{
	dispatch({type:"START_GET_ARTICLES"});
	// Define here Axios or Fetch Http requests
	var fakeArticlesResponse=articles.get();
	dispatch({type:"FINISH_GET_ARTICLES",payload:fakeArticlesResponse.data});
}