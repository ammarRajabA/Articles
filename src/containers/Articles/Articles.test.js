import React from 'react'
import { shallow ,mount } from 'enzyme';

import {Articles} from './Articles.component';
import fake_articles from '../../fakeAPI/fake_articles';

import articles from '../../fakeAPI/articles'

const test_props={
	match:{
		params:{
			id:"13245"
		}
	},
	articles:{
		articles:[],
		loading:false
	}
}

describe('Articles component',function () {

	it('match snapshot',function () {
		const component = mount(
			<Articles 
				articles={test_props.articles} 
				getArticles={()=>test_props.articles.articles=articles.get()}
			/>);
		
		const spy = jest.spyOn(Articles.prototype, "render");		
		expect(spy).toHaveBeenCalledTimes(0);

		component.setProps({...test_props});

		expect(spy).toHaveBeenCalledTimes(1);
		
		expect(component).toMatchSnapshot();
	})
})