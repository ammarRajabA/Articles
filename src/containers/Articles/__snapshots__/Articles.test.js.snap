// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Articles component match snapshot 1`] = `
<Articles
  articles={
    Object {
      "articles": Object {
        "data": Array [
          Object {
            "author": "Ammar Rajab",
            "content": "{\\"blocks\\":[{\\"key\\":\\"5viej\\",\\"text\\":\\"This is my first article, it so cool I can color words \\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[{\\"offset\\":49,\\"length\\":5,\\"style\\":\\"color-rgb(243,121,52)\\"}],\\"entityRanges\\":[],\\"data\\":{}},{\\"key\\":\\"dh0hv\\",\\"text\\":\\"I can also make bold italic or any other style.\\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[{\\"offset\\":16,\\"length\\":5,\\"style\\":\\"BOLD\\"},{\\"offset\\":21,\\"length\\":7,\\"style\\":\\"ITALIC\\"}],\\"entityRanges\\":[],\\"data\\":{}},{\\"key\\":\\"blt1q\\",\\"text\\":\\"I also can write math 25 =32\\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[{\\"offset\\":23,\\"length\\":2,\\"style\\":\\"SUPERSCRIPT\\"}],\\"entityRanges\\":[],\\"data\\":{}},{\\"key\\":\\"ddclt\\",\\"text\\":\\"ðŸ“£ Attention here : My Github account Ammar Rajab \\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[{\\"offset\\":2,\\"length\\":48,\\"style\\":\\"color-rgb(209,72,65)\\"}],\\"entityRanges\\":[{\\"offset\\":38,\\"length\\":11,\\"key\\":0}],\\"data\\":{}}],\\"entityMap\\":{\\"0\\":{\\"type\\":\\"LINK\\",\\"mutability\\":\\"MUTABLE\\",\\"data\\":{\\"url\\":\\"https://github.com/ammarrajaba\\",\\"targetOption\\":\\"_blank\\"}}}}",
            "id": "13245",
            "image": "https://picsum.photos/id/258/200/300",
            "title": "first",
          },
          Object {
            "author": "Ammar Rajab",
            "content": "{\\"blocks\\":[{\\"key\\":\\"35tp4\\",\\"text\\":\\"Another article by me,\\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[],\\"entityRanges\\":[],\\"data\\":{}},{\\"key\\":\\"a1v72\\",\\"text\\":\\"It shows how we align text\\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[],\\"entityRanges\\":[],\\"data\\":{}},{\\"key\\":\\"cifh6\\",\\"text\\":\\"Left\\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[],\\"entityRanges\\":[],\\"data\\":{}},{\\"key\\":\\"b5gsb\\",\\"text\\":\\"Middle\\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[],\\"entityRanges\\":[],\\"data\\":{\\"text-align\\":\\"center\\"}},{\\"key\\":\\"eopjr\\",\\"text\\":\\"and right\\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[],\\"entityRanges\\":[],\\"data\\":{\\"text-align\\":\\"right\\"}},{\\"key\\":\\"1h6bs\\",\\"text\\":\\"\\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[],\\"entityRanges\\":[],\\"data\\":{}},{\\"key\\":\\"d3mrc\\",\\"text\\":\\"and change font\\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[{\\"offset\\":0,\\"length\\":15,\\"style\\":\\"fontfamily-Impact\\"}],\\"entityRanges\\":[],\\"data\\":{}}],\\"entityMap\\":{}}",
            "id": "15555",
            "image": "https://picsum.photos/id/232/200/300",
            "title": "Second",
          },
          Object {
            "author": "Ammar Rajab",
            "content": "{\\"blocks\\":[{\\"key\\":\\"5viej\\",\\"text\\":\\"This is my first article, it so cool I can color words \\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[{\\"offset\\":49,\\"length\\":5,\\"style\\":\\"color-rgb(243,121,52)\\"}],\\"entityRanges\\":[],\\"data\\":{}},{\\"key\\":\\"dh0hv\\",\\"text\\":\\"I can also make bold italic or any other style.\\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[{\\"offset\\":16,\\"length\\":5,\\"style\\":\\"BOLD\\"},{\\"offset\\":21,\\"length\\":7,\\"style\\":\\"ITALIC\\"}],\\"entityRanges\\":[],\\"data\\":{}},{\\"key\\":\\"blt1q\\",\\"text\\":\\"I also can write math 25 =32\\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[{\\"offset\\":23,\\"length\\":2,\\"style\\":\\"SUPERSCRIPT\\"}],\\"entityRanges\\":[],\\"data\\":{}},{\\"key\\":\\"ddclt\\",\\"text\\":\\"ðŸ“£ Attention here : My Github account Ammar Rajab \\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[{\\"offset\\":2,\\"length\\":48,\\"style\\":\\"color-rgb(209,72,65)\\"}],\\"entityRanges\\":[{\\"offset\\":38,\\"length\\":11,\\"key\\":0}],\\"data\\":{}}],\\"entityMap\\":{\\"0\\":{\\"type\\":\\"LINK\\",\\"mutability\\":\\"MUTABLE\\",\\"data\\":{\\"url\\":\\"https://github.com/ammarrajaba\\",\\"targetOption\\":\\"_blank\\"}}}}",
            "id": "13445",
            "image": "https://picsum.photos/id/258/200/300",
            "title": "Third",
          },
          Object {
            "author": "Ammar Rajab",
            "content": "{\\"blocks\\":[{\\"key\\":\\"35tp4\\",\\"text\\":\\"Another article by me,\\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[],\\"entityRanges\\":[],\\"data\\":{}},{\\"key\\":\\"a1v72\\",\\"text\\":\\"It shows how we align text\\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[],\\"entityRanges\\":[],\\"data\\":{}},{\\"key\\":\\"cifh6\\",\\"text\\":\\"Left\\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[],\\"entityRanges\\":[],\\"data\\":{}},{\\"key\\":\\"b5gsb\\",\\"text\\":\\"Middle\\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[],\\"entityRanges\\":[],\\"data\\":{\\"text-align\\":\\"center\\"}},{\\"key\\":\\"eopjr\\",\\"text\\":\\"and right\\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[],\\"entityRanges\\":[],\\"data\\":{\\"text-align\\":\\"right\\"}},{\\"key\\":\\"1h6bs\\",\\"text\\":\\"\\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[],\\"entityRanges\\":[],\\"data\\":{}},{\\"key\\":\\"d3mrc\\",\\"text\\":\\"and change font\\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[{\\"offset\\":0,\\"length\\":15,\\"style\\":\\"fontfamily-Impact\\"}],\\"entityRanges\\":[],\\"data\\":{}}],\\"entityMap\\":{}}",
            "id": "15005",
            "image": "https://picsum.photos/id/232/200/300",
            "title": "Forth",
          },
        ],
      },
      "loading": false,
    }
  }
  getArticles={[Function]}
  match={
    Object {
      "params": Object {
        "id": "13245",
      },
    }
  }
>
  <div
    className="articles-container"
  >
    <Grid
      cols={3}
      handleSelect={[Function]}
      items={
        Object {
          "data": Array [
            Object {
              "author": "Ammar Rajab",
              "content": "{\\"blocks\\":[{\\"key\\":\\"5viej\\",\\"text\\":\\"This is my first article, it so cool I can color words \\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[{\\"offset\\":49,\\"length\\":5,\\"style\\":\\"color-rgb(243,121,52)\\"}],\\"entityRanges\\":[],\\"data\\":{}},{\\"key\\":\\"dh0hv\\",\\"text\\":\\"I can also make bold italic or any other style.\\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[{\\"offset\\":16,\\"length\\":5,\\"style\\":\\"BOLD\\"},{\\"offset\\":21,\\"length\\":7,\\"style\\":\\"ITALIC\\"}],\\"entityRanges\\":[],\\"data\\":{}},{\\"key\\":\\"blt1q\\",\\"text\\":\\"I also can write math 25 =32\\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[{\\"offset\\":23,\\"length\\":2,\\"style\\":\\"SUPERSCRIPT\\"}],\\"entityRanges\\":[],\\"data\\":{}},{\\"key\\":\\"ddclt\\",\\"text\\":\\"ðŸ“£ Attention here : My Github account Ammar Rajab \\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[{\\"offset\\":2,\\"length\\":48,\\"style\\":\\"color-rgb(209,72,65)\\"}],\\"entityRanges\\":[{\\"offset\\":38,\\"length\\":11,\\"key\\":0}],\\"data\\":{}}],\\"entityMap\\":{\\"0\\":{\\"type\\":\\"LINK\\",\\"mutability\\":\\"MUTABLE\\",\\"data\\":{\\"url\\":\\"https://github.com/ammarrajaba\\",\\"targetOption\\":\\"_blank\\"}}}}",
              "id": "13245",
              "image": "https://picsum.photos/id/258/200/300",
              "title": "first",
            },
            Object {
              "author": "Ammar Rajab",
              "content": "{\\"blocks\\":[{\\"key\\":\\"35tp4\\",\\"text\\":\\"Another article by me,\\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[],\\"entityRanges\\":[],\\"data\\":{}},{\\"key\\":\\"a1v72\\",\\"text\\":\\"It shows how we align text\\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[],\\"entityRanges\\":[],\\"data\\":{}},{\\"key\\":\\"cifh6\\",\\"text\\":\\"Left\\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[],\\"entityRanges\\":[],\\"data\\":{}},{\\"key\\":\\"b5gsb\\",\\"text\\":\\"Middle\\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[],\\"entityRanges\\":[],\\"data\\":{\\"text-align\\":\\"center\\"}},{\\"key\\":\\"eopjr\\",\\"text\\":\\"and right\\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[],\\"entityRanges\\":[],\\"data\\":{\\"text-align\\":\\"right\\"}},{\\"key\\":\\"1h6bs\\",\\"text\\":\\"\\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[],\\"entityRanges\\":[],\\"data\\":{}},{\\"key\\":\\"d3mrc\\",\\"text\\":\\"and change font\\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[{\\"offset\\":0,\\"length\\":15,\\"style\\":\\"fontfamily-Impact\\"}],\\"entityRanges\\":[],\\"data\\":{}}],\\"entityMap\\":{}}",
              "id": "15555",
              "image": "https://picsum.photos/id/232/200/300",
              "title": "Second",
            },
            Object {
              "author": "Ammar Rajab",
              "content": "{\\"blocks\\":[{\\"key\\":\\"5viej\\",\\"text\\":\\"This is my first article, it so cool I can color words \\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[{\\"offset\\":49,\\"length\\":5,\\"style\\":\\"color-rgb(243,121,52)\\"}],\\"entityRanges\\":[],\\"data\\":{}},{\\"key\\":\\"dh0hv\\",\\"text\\":\\"I can also make bold italic or any other style.\\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[{\\"offset\\":16,\\"length\\":5,\\"style\\":\\"BOLD\\"},{\\"offset\\":21,\\"length\\":7,\\"style\\":\\"ITALIC\\"}],\\"entityRanges\\":[],\\"data\\":{}},{\\"key\\":\\"blt1q\\",\\"text\\":\\"I also can write math 25 =32\\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[{\\"offset\\":23,\\"length\\":2,\\"style\\":\\"SUPERSCRIPT\\"}],\\"entityRanges\\":[],\\"data\\":{}},{\\"key\\":\\"ddclt\\",\\"text\\":\\"ðŸ“£ Attention here : My Github account Ammar Rajab \\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[{\\"offset\\":2,\\"length\\":48,\\"style\\":\\"color-rgb(209,72,65)\\"}],\\"entityRanges\\":[{\\"offset\\":38,\\"length\\":11,\\"key\\":0}],\\"data\\":{}}],\\"entityMap\\":{\\"0\\":{\\"type\\":\\"LINK\\",\\"mutability\\":\\"MUTABLE\\",\\"data\\":{\\"url\\":\\"https://github.com/ammarrajaba\\",\\"targetOption\\":\\"_blank\\"}}}}",
              "id": "13445",
              "image": "https://picsum.photos/id/258/200/300",
              "title": "Third",
            },
            Object {
              "author": "Ammar Rajab",
              "content": "{\\"blocks\\":[{\\"key\\":\\"35tp4\\",\\"text\\":\\"Another article by me,\\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[],\\"entityRanges\\":[],\\"data\\":{}},{\\"key\\":\\"a1v72\\",\\"text\\":\\"It shows how we align text\\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[],\\"entityRanges\\":[],\\"data\\":{}},{\\"key\\":\\"cifh6\\",\\"text\\":\\"Left\\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[],\\"entityRanges\\":[],\\"data\\":{}},{\\"key\\":\\"b5gsb\\",\\"text\\":\\"Middle\\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[],\\"entityRanges\\":[],\\"data\\":{\\"text-align\\":\\"center\\"}},{\\"key\\":\\"eopjr\\",\\"text\\":\\"and right\\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[],\\"entityRanges\\":[],\\"data\\":{\\"text-align\\":\\"right\\"}},{\\"key\\":\\"1h6bs\\",\\"text\\":\\"\\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[],\\"entityRanges\\":[],\\"data\\":{}},{\\"key\\":\\"d3mrc\\",\\"text\\":\\"and change font\\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[{\\"offset\\":0,\\"length\\":15,\\"style\\":\\"fontfamily-Impact\\"}],\\"entityRanges\\":[],\\"data\\":{}}],\\"entityMap\\":{}}",
              "id": "15005",
              "image": "https://picsum.photos/id/232/200/300",
              "title": "Forth",
            },
          ],
        }
      }
    >
      <div
        className="grid-container"
      />
    </Grid>
  </div>
</Articles>
`;
